### PR TITLE
Allow filtering apps with X-Parental-Controls desktop file entry

### PR DESCRIPTION
**Pull request by andrunko**
https://phabricator.endlessm.com/T26957

### CODE REVIEW

File: panels/search/cc-search-panel.c [hunk 1]
```diff
@@ -614,7 +614,7 @@ search_providers_discover_ready (GObject *source,
 
   for (l = discover_result->providers; l != NULL; l = l->next)
     {
-      g_autoptr(GFile) provider = l->data;
+      GFile *provider = l->data;
```
Comment (pwithnall) [inline panels/search/cc-search-panel.c]: This seems to have been a regression in the rebase of commit c773723d78, and should be squashed into that next time we rebase. I’ve checked, and the bug doesn’t seem to be present on the `eos3.6` branch.

File: panels/user-accounts/cc-app-permissions.c [hunk 2]
```diff
@@ -191,26 +192,54 @@ reload_apps (CcAppPermissions *self)
...
+        }
+      else if (g_desktop_app_info_has_key (G_DESKTOP_APP_INFO (app), "X-Parental-Controls"))
         {
-          g_debug (" → Skipping ‘%s’ due to seeing its flatpak ID already",
-                   g_app_info_get_id (app));
-          continue;
+          g_autofree gchar *parental_controls_type = NULL;
+          g_autofree gchar *executable = NULL;
+
+          parental_controls_type = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (app),
```
Comment (pwithnall) [inline panels/user-accounts/cc-app-permissions.c]: For future-proofness, maybe ignore desktop files which have `X-Parental-Controls=none`?

File: panels/search/cc-search-panel.c (hunk 1 above)
Comment (pwithnall) [reply panels/search/cc-search-panel.c]: (Now proposed in a separate PR as https://github.com/endlessm/gnome-control-center/pull/175.)

File: panels/user-accounts/cc-app-permissions.c (hunk 2 above)
Comment (andrunko) [reply panels/user-accounts/cc-app-permissions.c]: Done.

**Pull request merged**